### ISSUE TITLE
Using pkgfile defeats the purpose

### ISSUE THREAD

**Issue by yochananmarqos** [pkasemir/find-the-command#3]
If you came to that conclusion, you don't know how Pacman works.

**Comment by pkasemir** [comment]
If you don't want to use `pkgfile`, then uninstall it. it works just fine with `pacman`

**Comment by pkasemir** [comment]
`pkgfile` provides faster results, and thus a better user experience.

**Comment by yochananmarqos** [comment]
No, it doesn't. It duplicates something Pacman can already do. I used the original `find-the-command` program instead of `pkgfile` because it was faster. 

If your fork requires `pkgfile`, then... see title.

I'm done. Do whatever you want.

**Comment by pkasemir** [comment]
I don't know how `pkgfile` can be slower... in my testing it was always faster. The pkgfile DBs are much more efficient than the pacman DBs.

On top of that `fish` itself has decided to only use pkgfile because pacman is too slow. See https://github.com/fish-shell/fish-shell/issues/7841

**Comment by elvisisvan** [comment]
> No, it doesn't. It duplicates something Pacman can already do. I used the original `find-the-command` program instead of `pkgfile` because it was faster.
> 
> If your fork requires `pkgfile`, then... see title.
> 
> I'm done. Do whatever you want.

i get 
```
find-the-command: "nord" is not found locally, searching in repositories...
find-the-command: /var/cache/pkgfile/*.files are out of date, update? [Y/n] 
```
everytime i run a command even if it has just been updated right now, why is that?

![Image](https://github.com/user-attachments/assets/52d8043e-4300-4060-9f32-6eda3bb3466b)

**Comment by avktech78** [comment]
> > No, it doesn't. It duplicates something Pacman can already do. I used the original `find-the-command` program instead of `pkgfile` because it was faster.
> > If your fork requires `pkgfile`, then... see title.
> > I'm done. Do whatever you want.
> 
> i get
> 
> ```
> find-the-command: "nord" is not found locally, searching in repositories...
> find-the-command: /var/cache/pkgfile/*.files are out of date, update? [Y/n] 
> ```
> 
> everytime i run a command even if it has just been updated right now, why is that?
> 
> ![Image](https://github.com/user-attachments/assets/52d8043e-4300-4060-9f32-6eda3bb3466b)

for fish works this:
`function _cnf_need_to_update_files
    set dir "$argv[1]"
    set db_path (_cnf_pacman_db_path)

    # Check if there are any pacman db files
    if test (find "$db_path" -maxdepth 2 -type f -name "*.db" 2>/dev/null | wc -l) -eq 0
        # Prompt to refresh the pacman database if none found
        if _cnf_prompt_yn "No pacman db files in '$db_path', refresh?"
            _cnf_asroot pacman -Sy >&2
        else
            return 1
        end
    end

    # Check if there are any *.files in the target directory
    if test (find "$dir" -maxdepth 2 -type f -name "*.files" 2>/dev/null | wc -l) -eq 0
        set old_files all
    else
        # Get the newest .files and the newest pacman db
        set newest_files (/usr/bin/ls -t "$dir"/*.files | head -n 1)
        set newest_pacman_db (/usr/bin/ls -t "$db_path"/*.db | head -n 1)

        # Check if *.files is older than pacman db
        if test "$newest_files" -ot "$newest_pacman_db"
            _cnf_prompt_yn "$dir/*.files are out of date, update?"
            return $status
        end
    end

    # No update needed
    return 1
end
`

**Comment by bb10root** [comment]
> find-the-command: "nord" is not found locally, searching in repositories...
> find-the-command: /var/cache/pkgfile/*.files are out of date, update? [Y/n] 

for fish works this:

```
function _cnf_need_to_update_files
    set dir "$argv[1]"
    set db_path (_cnf_pacman_db_path)

    # Check if there are any pacman db files
    if test (find "$db_path" -maxdepth 2 -type f -name "*.db" 2>/dev/null | wc -l) -eq 0
        # Prompt to refresh the pacman database if none found
        if _cnf_prompt_yn "No pacman db files in '$db_path', refresh?"
            _cnf_asroot pacman -Sy >&2
        else
            return 1
        end
    end

    # Check if there are any *.files in the target directory
    if test (find "$dir" -maxdepth 2 -type f -name "*.files" 2>/dev/null | wc -l) -eq 0
        set old_files all
    else
        # Get the newest .files and the newest pacman db
        set newest_files (/usr/bin/ls -t "$dir"/*.files | head -n 1)
        set newest_pacman_db (/usr/bin/ls -t "$db_path"/*.db | head -n 1)

        # Check if *.files is older than pacman db
        if test "$newest_files" -ot "$newest_pacman_db"
            _cnf_prompt_yn "$dir/*.files are out of date, update?"
            return $status
        end
    end

    # No update needed
    return 1
end
```
